Installation.md

```diff
@@ -0,0 +1,267 @@
+```
+
+#### Switch to tmc user and login their shell ####
+
+```
+$ sudo su tmc
+$ bash -l
+$ cd
+```
+
+#### Import public key for rvm installation ####
+
+```
+$ command curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+$ rvm get stable
+```
+
+#### Setup user environment variables ####
+
+```
+$ echo "export GEM_HOME=$(ruby -e 'print Gem.user_dir')" >> ~/.bashrc
+$ echo "export PATH=$(echo $PATH):$GEM_HOME/bin" >> ~/.bashrc
+$ export GEM_HOME=$(ruby -e 'print Gem.user_dir')
+$ export PATH=$PATH:$GEM_HOME/bin
+```
+
+### PostgreSQL ###
+#### Create postgres user ####
+
+Run following command and when prompted give **tmc** as a password, if you want to use another username or stronger password change them into **config/database.yml**.
+
+```
+$ createuser tmc -s -P
+```
+
+If you cannot run above command, you can also create the user manually
+
+```
+$ sudo -s
+$ su - postgres
+$ psql
+    CREATE USER tmc WITH SUPERUSER PASSWORD 'tmc';
+```
+
+#### Set PostgreSQL to authenticate with md5 instead of local ####
+
+Locate the **pg_hba.conf** file with `locate pg_hba.conf`, the file should be located */etc/postgresql/<version number>/main/pg_hba.conf*. Open the file with your preferred text-editor, you need root privileges to edit the file. Around line 90 change
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     peer
+```
+to
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     md5
+```
+
+### TMC-server installation ###
+#### Clone the TMC repository ####
+
+```
+$ cd tmc-server
+$ git clone https://github.com/testmycode/tmc-server.git
+```
+:exclamation: If you are not using a github account, replace the repository submodule URLs with a HTTPS URL in .git/config eg. https://github.com/testmycode/tmc-checkstyle-runner.git
+```
+$ git submodule update --init --recursive
+$ gem install bundler && bundle install
+$ cp config/site.defaults.yml confg/site.yml
+$ env RAILS_ENV=production rake db:reset
+$ env RAILS_ENV=production rake assets:precompile
+```
+
+#### Build sandbox ####
+
+```
+$ cd ext/tmc-sandbox
+$ sudo make
+```
+
+#### Install tmc-sandbox web dependencies ####
+
+```
+$ cd web
+$ bundle install
+$ rake ext
+$ rvmsudo rake test
+```
+
+#### Compile rest of the externals ####
+
+```
+$ cd ../..
+$ rake compile
+```
+
+#### Install tmc-check ####
+
+```
+$ sudo apt-get install check
+$ cd tmc-sandbox/uml/output/tmc-check
+$ rvmsudo make rubygems install clean
+```
+
+#### Run the test suite ####
+In the tmc-server root directory run
+```
+$ rvmsudo rake spec
+```
+If you are using Java 8 one of the tests will fail.
+
+#### Enable pghero stats ####
+Add to postgresql.conf following lines
+```
+shared_preload_libraries = 'pg_stat_statements'
+pg_stat_statements.track = all
+```
+
+### Background services ###
```

Comment:
Maybe rename this section to "production setup" and move the necessary `env RAILS_ENV=prodcution rake ...` instructions things here.